### PR TITLE
Designed the server to serve the client.

### PR DESCRIPTION
**Pull request by Aditya7298**
* Express is used to write the server.
* Data is being saved in JSON files. There are 4 different JSON files to store info regarding : Users, Messages, Chat Rooms and Login Information. 
* Data Model : Screenshot 2021-08-01 at 11 41 09 PM

### CODE REVIEW

File: routes/loginRoutes.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+router.post("/", (req, res) => {
+  const { username, password } = req.body;
+  loginController
+    .authenticateUser(username, password)
+    .then((data) => {
+      res.status(200).json(data);
+    })
+    .catch((err) => {
+      const { code, message } = err;
```
Comment (ballerbuoy) [inline routes/loginRoutes.js:17]: it is possible for your code to be undefined, when loginController rejects from catch block

File: controllers/UserController.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      }
+    });
+  }
+
+  getAllUsers() {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const userDataJSON = await this.readFromDB();
+        const userData = JSON.parse(userDataJSON);
+        resolve(JSON.stringify(userData));
```
Comment (ballerbuoy) [inline controllers/UserController.js:35]: can pass useDataJSON instead of stringifying the data again

Comment (Aditya7298) [reply controllers/UserController.js:35]: noted

File: controllers/ChatRoomController.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+        resolve(chatRoomData[chatRoomId]);
+      } catch (err) {
+        reject({ ...err });
+      }
+    });
+  }
+
+  createChatRoom(payload) {
+    return new Promise(async (resolve, reject) => {
```
Comment (hetpatel33) [inline controllers/ChatRoomController.js:31]: Can we not make createChatRoom async directly instead of a manual promise?

Comment (hetpatel33) [reply controllers/ChatRoomController.js:31]: Check for each such occurrences

File: controllers/utils.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const checkPayloadForKeys = (payload, keys) => {
+  if (!payload) {
+    return false;
+  }
+
+  return keys.reduce((result, key) => {
+    return result && Object.keys(payload).includes(key);
```
Comment (hetpatel33) [inline controllers/utils.js:7]: Can we not early return if one of this results in falsy value?

**Review by hetpatel33 — CHANGES_REQUESTED** [review]

**Pull request merged**